routemaster/tests/test_feeds.py

```diff
@@ -0,0 +1,72 @@
+import mock
+import pytest
+import httpretty
+
+from routemaster.feeds import Feed, FeedNotFetched, feeds_for_state_machine
+from routemaster.config import Feed as FeedConfig
+from routemaster.config import Gate, NoNextStates, StateMachine
+from routemaster.exit_conditions import ExitConditionProgram
+
+
+def test_feeds_for_state_machine():
+    state_machine = StateMachine(
+        name='example',
+        feeds=[
+            FeedConfig(name='test_feed', url='http://localhost/<label>'),
+        ],
+        states=[
+            Gate(
+                name='start',
+                triggers=[],
+                next_states=NoNextStates(),
+                exit_condition=ExitConditionProgram('false'),
+            ),
+        ]
+    )
+
+    feeds = feeds_for_state_machine(state_machine)
+
+    assert 'test_feed' in feeds
+    assert feeds['test_feed'].data is None
+    assert feeds['test_feed'].url == 'http://localhost/<label>'
+    assert feeds['test_feed'].state_machine == 'example'
+
+
+@httpretty.activate
+def test_fetch_only_once():
+    httpretty.register_uri(
+        httpretty.GET,
+        'http://example.com/test_machine/label1',
+        body='{"foo": "bar"}',
+        content_type='application/json',
+    )
+
+    feed = Feed('http://example.com/<state_machine>/<label>', 'test_machine')
+
+    with mock.patch('requests.Response.json') as json:
+        feed.fetch('label1')
+        feed.fetch('label1')
+        feed.fetch('label1')
+
+        assert json.call_count == 1
+
+
+@httpretty.activate
+def test_lookup():
```

Comment:
I think this test is a duplicate of `test_accesses_variable_in_feed`

Reply:
Not quite, `test_accesses_variable_in_feed` goes through a layer of dynamic dispatch (in a way) in the context, which is worth testing I think.